### PR TITLE
Add test for createPr

### PR DESCRIPTION
**Pull request by thiagodsti**
- Add test for create PR
- Add .env support for tests

### CODE REVIEW

File: src/test/kotlin/com/github/jensim/megamanipulator/TestHelper.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.github.jensim.megamanipulator.settings.CodeHostSettings.GitHubSettings.Companion.GITHUB_TOKEN
+import com.github.jensim.megamanipulator.settings.CodeHostSettings.GitHubSettings.Companion.GITHUB_USERNAME
+import com.github.jensim.megamanipulator.settings.ForkSetting
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.util.Properties
+
+class TestHelper {
+
+    companion object {
```
Comment (jensim) [inline src/test/kotlin/com/github/jensim/megamanipulator/TestHelper.kt:13]: Class with all members in companion object can be declared as object

File: src/test/kotlin/com/github/jensim/megamanipulator/TestHelper.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        const val MEGA_MANIPULATOR_REPO = "mega-manipulator"
+        fun getGithubCredentials(): CodeHostSettings.GitHubSettings {
+            val prop = Properties()
+            return try {
+                val fis = FileInputStream(".env")
+                prop.load(fis)
+                val username = prop.getProperty(GITHUB_USERNAME)
```
Comment (jensim) [inline src/test/kotlin/com/github/jensim/megamanipulator/TestHelper.kt:20]: Lazy delegate?
Fallback to sys env vars?

File: src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt [hunk 3]
```diff
@@ -228,5 +228,10 @@ sealed class CodeHostSettings
         init {
             validate()
         }
+
+        companion object {
+            const val GITHUB_USERNAME = "GITHUB_USERNAME"
```
Comment (jensim) [inline src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt:233]: These consts dont belong in main code?

File: src/test/kotlin/com/github/jensim/megamanipulator/actions/vcs/githubcom/GithubComClientTest.kt [hunk 4]
```diff
@@ -1,35 +1,39 @@
...
 
 class GithubComClientTest {
 
-    private val githubSettings = GitHubSettings(
-        username = System.getenv("GITHUB_USERNAME") ?: "jensim",
-        forkSetting = PLAIN_BRANCH,
-    )
-    private val password = System.getenv("GITHUB_TOKEN")
+    private val githubSettings = TestHelper.getGithubCredentials()
+    private val password = System.getenv(GITHUB_TOKEN) ?: System.getProperty(GITHUB_TOKEN)
```
Comment (jensim) [inline src/test/kotlin/com/github/jensim/megamanipulator/actions/vcs/githubcom/GithubComClientTest.kt:36]: Shouldn't prop have presidence over env?

File: src/test/kotlin/com/github/jensim/megamanipulator/actions/vcs/githubcom/GithubComClientTest.kt [hunk 5]
```diff
@@ -58,6 +62,32 @@ class GithubComClientTest {
         json = SerializationHolder.instance.readableJson
     )
 
+    @Test
```
Comment (jensim) [inline src/test/kotlin/com/github/jensim/megamanipulator/actions/vcs/githubcom/GithubComClientTest.kt:65]: Maybe this test should should be part of a bigger flow. 
IntegrationTest like
Search
Clone
Apply changes
Commit & push (with lazy fork being created)
Draft PR
Reword PR
Decline PR
Remove Fork

File: src/test/kotlin/com/github/jensim/megamanipulator/TestHelper.kt [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.github.jensim.megamanipulator.settings.CodeHostSettings
+import com.github.jensim.megamanipulator.settings.CodeHostSettings.GitHubSettings.Companion.GITHUB_TOKEN
+import com.github.jensim.megamanipulator.settings.CodeHostSettings.GitHubSettings.Companion.GITHUB_USERNAME
+import com.github.jensim.megamanipulator.settings.ForkSetting
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.util.Properties
+
+class TestHelper {
```
Comment (jensim) [inline src/test/kotlin/com/github/jensim/megamanipulator/TestHelper.kt:11]: Nice to have .env support 🥳

File: src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt (hunk 3 above)
Comment (thiagodsti) [reply src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt:233]: It's in the MySettings

File: src/test/kotlin/com/github/jensim/megamanipulator/actions/vcs/githubcom/GithubComClientTest.kt (hunk 5 above)
Comment (thiagodsti) [reply src/test/kotlin/com/github/jensim/megamanipulator/actions/vcs/githubcom/GithubComClientTest.kt:65]: Agree

File: src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt (hunk 3 above)
Comment (jensim) [reply src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt:233]: What I mean is, the prop should never be used in the "prod" code

Comment (thiagodsti) [reply src/main/kotlin/com/github/jensim/megamanipulator/settings/MySettings.kt:233]: I got your point, it makes sense

**Pull request closed without merging**